server/container/lifecycle-api/src/test/java/org/apache/james/lifecycle/api/LeakAwareFixture.java

```diff
@@ -0,0 +1,36 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.lifecycle.api;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+
+public class LeakAwareFixture {
+
+    // using reflect to change LeakAware.LEVEL value
+    public static void forceChangeLevel(Disposable.LeakAware.Level level) throws NoSuchFieldException, IllegalAccessException {
+        final Field field = Disposable.LeakAware.class.getDeclaredField("LEVEL");
+        field.setAccessible(true);
+        final Field modifiersField = Field.class.getDeclaredField("modifiers");
+        modifiersField.setAccessible(true);
+        modifiersField.setInt(field, field.getModifiers() & ~Modifier.FINAL);
+        field.set(null, level);
+    }
```

Comment:
I prefer us to pass the system property to the surefire plugin rather than having reflection in the middle of production code.

Reply:
(y), let me override it in pom.xml